packages/firestore/src/core/firestore_client.ts

```diff
@@ -590,6 +590,10 @@ export class FirestoreClient {
     return this.databaseInfo.databaseId;
   }
 
+  isShutdown(): boolean {
```

Comment:
You could make this look like a readonly property to callers by marking this a getter.

See e.g. how `_started` is handled in `MemoryPersistence`